api/src/main/java/org/openmrs/module/registrationcore/api/impl/PatientCreatedListener.java

```diff
@@ -0,0 +1,129 @@
+        return actions;
+    }
+
+    /**
+     * Exports patient to MPI, and updates the local patient
+     * with the new patient identifier generated by the MPI.
+     *
+     * @param message message with properties.
+     */
+    @Override
+    public void performMpiAction(Message message) {
+        Patient patient = extractPatient(message);
+        if (patient != null){
+            PatientIdentifierType mpiPatientIdType = patientService.getPatientIdentifierTypeByUuid(mpiProperties.getMpiPersonIdentifierTypeUuid());
+            PatientIdentifier mpiPatientId = patient.getPatientIdentifier(mpiPatientIdType);
+            if (mpiPatientId != null && !mpiPatientId.getIdentifier().isEmpty()) {
+                // Patient originated in MPI and thus should not be pushed to the MPI
+                // TODO does mpi have openmrs id, if no update, if yes stop
+                return;
+            }else{
+                String mpiPatientIdString = null;
+                try {
+                    mpiPatientIdString = pushPatientToMpiServer(patient);
+                } catch (Exception e) {
+                    ErrorHandlingService errorHandler = coreProperties.getPixErrorHandlingService();
+                    if (errorHandler == null) {
+                        throw new MpiException("PIX patient push exception occurred "
+                                + "with not configured PIX error handler", e);
+                    } else {
+                        errorHandler.handle(prepareParameters(patient),
+                                PixErrorHandlingService.SENDING_PATIENT_AFTER_PATIENT_CREATION_DESTINATION,
+                                true,
+                                ExceptionUtils.getFullStackTrace(e));
+                        throw new MpiException("PIX patient push exception occurred", e);
+                    }
+                }
+                if (mpiPatientIdString != null) {
+                    updatePatient(patient, mpiPatientIdString);
+                }
+            }
+        }else{
+            throw new MpiException("PeformMpiAction error: extractPatient returned null patient");
+        }
+    }
+
+    /**
+     * Pushes patient to MPI using MPI provider specified in global properties
+     *
+     * @param patient patient to be pushed to MPI
+     * @return MPI global identifier of the patient, generated by the MPI
+     */
+    private String pushPatientToMpiServer(Patient patient) {
+        return coreProperties.getMpiProvider().exportPatient(patient);
+    }
+
+    /**
+     * Updates patient in local DB with MPI global identifier generated by the MPI.
+     *
+     * @param patient patient to be updated
+     * @param mpiPatientId MPI global identifier for the provided patient
+     */
+    private void updatePatient(final Patient patient, final String mpiPatientId) {
+        User creator = patient.getCreator();
+        patient.addIdentifier(createPatientIdentifier(creator, mpiPatientId));
```

Comment:
Don't set creator here, the service layer automatically sets it to the authenticated user, I believe this functionality should run in a daemon thread because it's automated, isn't it?

Reply:
I will experiment to see if the creator is set automatically

Reply:
I tried not setting the creator and the result was that the creator was set to the Daemon user instead of the authenticated user, I think I will leave it as it is for now, but I can create a ticket to investigate why the service layer is not setting it to the authenticated user and work on this once we get through the merge request?

Reply:
And the correct behavior is the creator should be set to daemon user because this code runs inside a daemon thread and there is no authenticated user.

Reply:
It's not a bug, it's the correct behavior, so don't set the creator.

Reply:
Ah I see okay my bad, I will correct it to not set the creator then, thanks